packages/core/strapi/lib/core/loaders/plugins/get-enabled-plugins.js

```diff
@@ -26,12 +28,19 @@ const toDetailedDeclaration = declaration => {
   }
   if (has('resolve', declaration)) {
     let pathToPlugin = '';
-    try {
-      pathToPlugin = dirname(require.resolve(declaration.resolve));
-    } catch (e) {
-      if (existsSync(declaration.resolve) && statSync(declaration.resolve).isDirectory()) {
-        pathToPlugin = declaration.resolve;
-      } else {
+
+    if (isPath(declaration.resolve)) {
+      pathToPlugin = isAbsolute(declaration.resolve)
+        ? declaration.resolve
+        : join(strapi.dir, declaration.resolve);
```

Comment:
You could use [`path.resolve`](https://nodejs.org/api/path.html#path_path_resolve_paths) to get rid of the ternary conditional:

```js
pathToPlugin = path.resolve(strapi.dir, declaration.resolve);
```

If `declaration.resolve` is an absolute path, it'll be used as is. If it's a relative one, it'll be relative to `strapi.dir`.

Reply:
Good idea